### ISSUE TITLE
[Feature Request]: Some GS hardware fixes

### ISSUE THREAD

**Issue by ghost** [PCSX2/pcsx2#5852]
### Description

Hi! I would like to contribute to the GS hardware database, letting you know which ones hacks i use for the games.

Crash Bandicoot - The Wrath Of Cortex | SLUS-20238 | (**Deinterlace** in **Bob TFF**) in this game with this option enabled, fixes the issue of the screen flickering all time of gameplay.
.
[EmuCore/GS]
interlace = 3
.
.
.
Grand Theft Auto - Liberty City Stories | SLUS-21423 | (**GPU Palette Conversion** enabled) in this game when activating this option the performance is improved, slightly lowering the FPS drop.
.
[EmuCore/GS]
paltex = true 
.
.
.
Madagascar | SLUS-21015 | (**Half Pixel Offset** in **Special Texture (Agressive)**);**(Round Sprite** in **Full**);(**Merge Sprite** enabled) with these three configurations combined, fixes the game's blur issue, keeping the game with correct blur distance and visuals present on the game field and also correct the game lights positioning correctly.
.
[EmuCore/GS]
UserHacks = true
UserHacks_HalfPixelOffset = 3
UserHacks_round_sprite_offset = 2
UserHacks_merge_pp_sprite = true

![image](https://user-images.githubusercontent.com/84431888/162854436-065cb45a-6400-4f1c-a5f1-fd20a7d43647.png)
![image](https://user-images.githubusercontent.com/84431888/162854513-d8425af3-8b7e-494e-9b63-696ecd986e59.png)
.
.
.
LEGO Star Wars II - The Original Trilogy | SLUS-21409 | (**Textures Offset** in **X: 300 Y: 300**) In this game with these settings, it does not have the "ghost" effect.
.
[EmuCore/GS]
UserHacks = true
UserHacks_TCOffsetX = 300
UserHacks_TCOffsetY = 300

![image](https://user-images.githubusercontent.com/84431888/162829308-476695d1-3abe-46d8-b6f2-d75587b65c9a.png)
![image](https://user-images.githubusercontent.com/84431888/162829482-b01c09e5-6d2d-4b6e-b351-c764e0c9b506.png)
.
.
.
Ratatouille | SLUS-21541 | (**Half Pixel Offset** in **Special Texture (Agressive)**);(**Textures Offset** in **X: 300 Y: 300**) in this game with these two configurations combined, it does not have the "ghost" effect.
.
[EmuCore/GS]
UserHacks = true
UserHacks_HalfPixelOffset = 3
UserHacks_TCOffsetX = 450
UserHacks_TCOffsetY = 450

![image](https://user-images.githubusercontent.com/84431888/162854629-2a72c474-3172-4663-910c-c3b907ea42c1.png)
![image](https://user-images.githubusercontent.com/84431888/162854721-0c5236a5-9faa-4225-8228-f9a2fbd9f730.png)
.
.
.
### Reason

Add these games in the GS hardware hacks fixes database

**Comment by refractionpcsx2** [comment]
UserHacks_TCOffset doesn't exist in the GameDB because it varies depending on what upscale you use, and not everybody is going to have the same upscale as you.

Can you also label what the fixes do for each game please.

**Comment by SoloBackup** [comment]
I understand, thanks for the reply! I edited the post with the corrections of the information of what each hack does in the game.

**Comment by refractionpcsx2** [comment]
Just to note, your changes to Madagascar don't "Fix" the lighting, they remove it (look at the window on the left), this is probably merge sprite, how does it look without it?

**Comment by SoloBackup** [comment]
Ah really, I hadn't noticed this detail. Without this lighting, the game takes place normally, but I don't know if it will be valid to create a hack fix for this, since a detail of the game is lost.
Of the ones I listed so I think only Crash will be used, because the GTA Liberty City Stories one improves the game's performance but not 100% so it's debatable whether it's worth adding to the database or not. LEGO and Ratatouille vary with the upscale chosen by each person, so I also believe that adding automatic hacks to the database would not be valid. And the case of Madagascar. Only Crash Wrath Of Cortex is sure that it was a problem fixed and that no details were lost (but I only tested the ntsc version of the game, the European one I don't know how the hack works) so that's it I hope I helped in some way, and thanks for the guidance, there really are details that I hadn't noticed, thank you!
